plugins/modules/content_upload.py

```diff
@@ -174,7 +174,7 @@ def main():
 
                 with open(b_src, 'rb') as contentfile:
                     for chunk in iter(lambda: contentfile.read(CONTENT_CHUNK_SIZE), b""):
-                        data = {'content': chunk, 'offset': offset, 'size': len(chunk)}
+                        data = {'content': chunk, 'offset': offset, 'size': size}
```

Comment:
Did you check, this does not blow up if you chunk the file into more that two pieces?

Reply:
Yes, I tested it with the file from #1043 which is 6 (well, 5.8) chunks big

Reply:
Strange but ok.
I think the actual bug is here:
https://github.com/Katello/katello/blob/540e695c5e108df99d84acae620ab641b6c898f0/app/services/katello/pulp3/content.rb#L32
The size parameter for the content_range should be taken from the upload object, not the update_upload request.

Reply:
that said again, pulp upload does not even use or expect the size field (it's marked read_only, but never actually set).

Reply:
Yeah, I also wondered why the chunk upload needs the size at all, but oh well… I'll open a refactor issue against katello

Reply:
https://projects.theforeman.org/issues/31300